### PR TITLE
Remove all user code paths to manually add repositories

### PR DESCRIPTION
**Pull request by nicksnyder**
This addresses the first checkbox in https://github.com/sourcegraph/sourcegraph/issues/914#issuecomment-440067022

The only supported way to add repositories to Sourcegraph is by configuring external services.

This does not impact auto enablement on sourcegraph.com because that code path does not go through the graphql api:
https://sourcegraph.com/github.com/sourcegraph/sourcegraph/-/blob/cmd/frontend/backend/repos.go#L69:12

### CODE REVIEW

File: web/src/repo/RepositoryErrorPage.tsx [hunk 1]
```diff
@@ -251,28 +208,10 @@ export class RepositoryErrorPage extends React.PureComponent<Props, State> {
                                         {this.state.canAddOrError === true && (
                                             <>
                                                 As a site admin, you can add this repository to Sourcegraph to allow
-                                                users to search and view it.
+                                                users to search and view it by updating your site configuration.
```
Comment (beyang) [inline web/src/repo/RepositoryErrorPage.tsx]: Can we be more prescriptive here by giving an example of such configuration or linking to the appropriate docs?

Comment (nicksnyder) [reply web/src/repo/RepositoryErrorPage.tsx]: The best current doc page that I could find was https://sourcegraph.com/help/admin (specifically the bullet list called "Integrations").

**Comment by nicksnyder** [comment]
> This change will make it higher friction to add public repositories from github.com, which seems like it's fairly important for new users who want to test out their Sourcegraph instance. What's the justification that this okay/acceptable?

I think the short answer is that this is acceptable for 3.0-preview and will be resolved by 3.0.

> The real goal seems to be to get rid of "loose" repositories that are named in a way that's unconstrainted by any code host connection. The removal of the "Add a public repository" UI seems like a side effect. If that UI is still valuable, could we instead have some sort of implicit or default github.com GitHubConnection that covers those repos?

Yes, this is one possible solution that we can pursue for 3.0, but I am not confident will be done in time for 3.0-preview.

The general principle here is that we are ok deleting stuff for 3.0-preview so that the delta between 3.0-preview and 3.0 is merely adding things (instead of changing behavior).

File: cmd/frontend/graphqlbackend/schema.graphql [hunk 2]
```diff
@@ -49,25 +49,6 @@ type Mutation {
     updateExternalService(input: UpdateExternalServiceInput!): ExternalService!
```
Comment (tsenart) [inline cmd/frontend/graphqlbackend/schema.graphql]: Archeology question: What led us to have the GraphQL schema duplicated across this file and `schema.go`?

**Pull request merged**